app/controllers/split_checkout_controller.rb

```diff
@@ -204,6 +203,7 @@ def shipping_method_ship_address_not_required?
 
   def process_voucher
     if add_voucher
+      VoucherAdjustmentsService.calculate(@order)
```

Comment:
As long as taxes don't change after this it should be fine. If I followed correctly taxes are recalculated if address changes, and also after the order move to the "payment" step, which all happen before we can reach this code, is that correct ?